### ISSUE TITLE
teleport-kube-agent 8.1.1 gets "malformed HTTP response" for kubectl exec 

### ISSUE THREAD

**Issue by mdaniel** [gravitational/teleport#10109]
### Description

**What happened**:

```console
$ kubectl exec -it $my_pod -- bash
error: Internal error occurred: error executing command in container: Internal error occurred: error executing command in container: error sending request: Post "https://172.20.0.1:443/api/v1/namespaces/my-ns/pods/my-pod-59fbc6ff7d-7xcv4/exec?command=bash&container=c-name&stdin=true&stdout=true&tty=true": malformed HTTP response "\x00\x00\x18\x04\x00\x00\x00\x00\x00\x00\x05\x00\x04\x00\x00\x00\x03\x00\x00\x00\xfa\x00\x06\x00\x10\x01@\x00\x04\x00\x04\x00\x00"
```
and the message is confirmed in the teleport-agent pod logs

**What you expected to happen**:

Successful exec into the container

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Upgrade [teleport-kube-agent](https://github.com/gravitational/teleport/blob/v8.1.1/examples/chart/teleport-kube-agent/Chart.yaml#L3) to 8.1.1 from 8.1.0
2. Try and exec

### Server Details
- Teleport version (run `teleport version`): presumably 8.1.1
- Server OS (e.g. from `/etc/os-release`): N/A
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS on EKS 1.21
- Additional details:

### Client Details
- Tsh version (`tsh version`): `Teleport v8.1.1 git: go1.17.6`
- Computer OS (e.g. Linux, macOS, Windows): Linux
- Browser version (for UI-related issues): N/A
- Installed via (e.g. apt, yum, brew, website download): `brew`
- Additional details:

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)

**Comment by RaphaelNeumann** [comment]
I had the same issue with a new fresh install of 8.1.1

**Comment by xacrimon** [comment]
This is fixed by #10011 and will be released soon in a patch release.

**Comment by programmerq** [comment]
8.1.3 fixed this, so I'm closing this issue.
